### PR TITLE
feature: add e2e and distribution policy

### PR DESCRIPTION
**Pull request by qiuming520**
#### What type of PR is this?

/kind feature 
#### What does this PR do?

add distribution policy code and  e2e 
#### Which issue(s) does this PR fix?

Fixes #https://github.com/kosmos-io/kosmos/issues/297
Fixes #https://github.com/kosmos-io/kosmos/issues/62
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by duanmengkk** [comment]
what is the problem with E2E?

**Comment by qiuming520** [comment]
> what is the problem with E2E?

It may be due to insufficient resources that the cluster cannot be created after several attempts.

**Comment by duanmengkk** [comment]
> > what is the problem with E2E?
> 
> It may be due to insufficient resources that the cluster cannot be created after several attempts.

PLZ confirm it

**Comment by qiuming520** [comment]
> > > what is the problem with E2E?
> > 
> > 
> > It may be due to insufficient resources that the cluster cannot be created after several attempts.
> 
> PLZ confirm it

@duanmengkk e2e has been fixed， help merge plz !

File: pkg/scheduler/lifted/plugins/leafnodedistribution/leafnode_distribution_policy.go [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+
+	kosmosv1alpha1 "github.com/kosmos.io/kosmos/pkg/apis/kosmos/v1alpha1"
+	"github.com/kosmos.io/kosmos/pkg/generated/clientset/versioned"
+	dpInformer "github.com/kosmos.io/kosmos/pkg/generated/informers/externalversions"
```
Comment (duanmengkk) [inline pkg/scheduler/lifted/plugins/leafnodedistribution/leafnode_distribution_policy.go:17]: It‘s not necessary to use alias here?
```suggestion
"github.com/kosmos.io/kosmos/pkg/generated/informers/externalversions"
```

**Pull request merged**